integrationTests/ts-esm/package.json

```diff
@@ -0,0 +1,15 @@
+{
+  "private": true,
+  "description": "graphql-js ESM should compile with all supported TS versions",
+  "scripts": {
+    "test": "node test.js"
+  },
+  "dependencies": {
+    "graphql": "file:../graphql-esm.tgz",
+    "typescript-4.2": "npm:typescript@4.2.x",
```

Comment:
For some reason Typescript@4.1.x didn't work, I'm not completely sure the reason :man_shrugging: